### ISSUE TITLE
[runtime env] using wrong ray version in pip packages breaks runtime envs forever

### ISSUE THREAD

**Issue by AmeerHajAli** [ray-project/ray#16277]
if you do something like this:
```
a = TestActor.options(name="abc", lifetime="detached", runtime_env={"pip": [
        "chess",
        "opentelemetry-api==1.0.0rc1",
        "ray==1.4.0rc1",
    ]}).remote()
```
but the server has for example `ray==1.4.0rc2`, it will break. which is the correct behavior. BUT, the worker that failed to launch continues to relaunch indefinitely. So even if you rerun the code with a completely new code or remove the faulty line, the worker continuously prints:
```
(raylet, ip=172.31.127.228)     import ray
(raylet, ip=172.31.127.228) ModuleNotFoundError: No module named 'ray'
(raylet, ip=172.31.127.228) Traceback (most recent call last):
(raylet, ip=172.31.127.228)   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/workers/default_worker.py", line 8, in <module>
(raylet, ip=172.31.127.228)     import ray
(raylet, ip=172.31.127.228) ModuleNotFoundError: No module named 'ray'
(raylet, ip=172.31.127.228) Traceback (most recent call last):
(raylet, ip=172.31.127.228)   File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/workers/default_worker.py", line 8, in <module>
```

### What is the problem?

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by AmeerHajAli** [comment]
cc @edoakes.
I do not think we need to fix this for 1.4. It seems like less critical.

**Comment by architkulkarni** [comment]
Closed by https://github.com/ray-project/ray/pull/17249
